Purchases/Purchasing/OfferingsManager.swift

```diff
@@ -96,19 +98,28 @@ private extension OfferingsManager {
 
     func handleOfferingsBackendResult(with data: [String: Any], completion: ((Offerings?, Error?) -> Void)?) {
         let productIdentifiers = extractProductIdentifiers(fromOfferingsData: data)
+        guard !productIdentifiers.isEmpty else {
+            self.handleOfferingsUpdateError(ErrorUtils.configurationError(), completion: completion)
+            return
+        }
```

Comment:
I don't think we should take this change lightly. We have traditionally returned nil offerings when there's a configuration error. It's probably one of the biggest support questions we get. It's not that we don't agree with returning an error instead, but I just wanted to mention that we did this by design, and it was in my opinion a bad design decision.

Reply:
I think returning `nil` from the backend makes sense, but then moving into Swift land, we probably want to attach clear meaning to it with a typed error. If we were to leave it as nil, I don't see any benefit. This way, folks have a concrete place to look for an answer. I recognize that I'm also probably missing context... so if I'm missing any nuances, sorry!

Reply:
I think it's time to make this change. We have more context on https://app.shortcut.com/revenuecat/story/7885/return-error-in-getofferings-if-no-products-are-fetched. There's a full epic for this issue lol

Reply:
me, seeing my own comments on that ticket from literally 2 months ago: 
![I have no memory of this place](https://media.giphy.com/media/FPjbHO0jJxGsE/giphy.gif?cid=ecf05e47qts4eicdguymwloj3mg3b6980dcrd1z6kom4nbc7&rid=giphy.gif&ct=g)

Reply:
I'm tagging @rkotzy for thoughts on this. 

I 100% agree with @taquitos, I can't see any benefit to just returning empty, since there's no use case where having empty offerings.current isn't caused by a configuration error. 

In this PR, this change in behavior would be for v4.0.0, but I'd like to also ship it to 3.x.x.

Reply:
@aboedo does this return an error if there's all Offerings are nil, or just if offerings.current is nil? There could be a case where we break existing setups if an error is now returned if just offerings.current is nil, but some of the other offerings exist